lib/src/Collection.cpp

```diff
@@ -946,6 +1012,42 @@ void Collection::ExecuteStoredProcedure(
 	ExecuteStoredProcedureAsync(resource_id, input).get();
 }
 
+void Collection::ExecuteStoredProcedureWithOutput(

```

Comment:
I didn't know for this, cool! Does this really warrants new function, can we just reuse existing one, with output as optional param?

Reply:
I used it like this.

==== azure store procedure ==== 
function spTest(tv) {
    var collection = getContext().getCollection();
    var collectionLink = collection.getSelfLink();
	var prefix = "";
	var retDoc = {};
	retDoc.a = "a";
	retDoc.b = 17;
	getContext().getResponse().setBody(prefix + JSON.stringify(retDoc));
}

=== function declear ===
 void Collection::ExecuteStoredProcedureWithOutput(
	const utility::string_t& resource_id,
	const web::json::value& input,
	web::json::value& output) const

=== consumer code ===
value inputObj;
inputObj[U("num")] = value::number(0);

value inputDoc, output;
inputDoc[0] = inputObj;

try {
	coll->ExecuteStoredProcedureWithOutput(proc->resource_id(), inputDoc, output);
	LOGI << output.serialize();
}
catch (DocumentDBRuntimeException ex) {
	LOGE << "ERR MSG : " << ex.message();
	return false;
}

Reply:
No, my questions is more - is this same REST call as with `ExecuteStoredProcedureAsync`, maybe we can reuse that function instead of creating new, almost similar?

BTW, you can add this test to tests, I would'nt mind:)